### PR TITLE
AFC-1659-Deviationreport fixes

### PR DESCRIPTION
**Pull request by nils-vertel**
* Fix missing field motivering in send
* Replaced dummy values with real values
* PEP8 fixes

### CODE REVIEW

**Comment by arturchilangwa-vertel** [comment]
@nils-vertel or @anderswallenquist 
Please fix conflicts these conflicts

File: outplacement/models/outplacement.py [hunk 1]
```diff
@@ -185,49 +184,48 @@ def _track_template(self, tracking):
...
+    #             outplacement._message_add_suggested_recipient(
+    #                 recipients,
+    #                 partner=outplacement.partner_id,
+    #                 reason=_('Contact'))
+    #         elif outplacement.email_from:
+    #             outplacement._message_add_suggested_recipient(
+    #                 recipients,
+    #                 email=outplacement.email_from,
+    #                 reason=_('Contact Email'))
+    #     return recipients
```
Comment (arturchilangwa-vertel) [inline outplacement/models/outplacement.py:228]: Please give a reason why this is commented or remove it.

File: outplacement_deviation_report/wizards/deviation_report_wizard.py [hunk 2]
```diff
@@ -58,49 +86,68 @@ def float_to_time(self, value):
...
-                    "starttid": self.expected_time_start,
-                    "sluttid": self.expected_time_end,
+                    "starttid": self.float_to_time(self.expected_time_start),
+                    "sluttid": self.float_to_time(self.expected_time_end),
+                    },
+                "motivering": self.deviation_reason or ''
                 },
-            }
         }
+        _logger.warn(pprint.pformat(payload))
```
Comment (arturchilangwa-vertel) [inline outplacement_deviation_report/wizards/deviation_report_wizard.py]: Do we need to log the payload?

File: outplacement_deviation_report/wizards/deviation_report_wizard.py [hunk 3]
```diff
@@ -110,5 +157,10 @@ def action_send(self):
             payload=json.dumps(payload),
             headers=api.get_headers(),
             params=querystring)
-        if response.status_code != 200:
+        if response.status_code not in (200, 201):
+            _logger.warning(response.text)
+            _logger.warning(response.status_code)
```
Comment (arturchilangwa-vertel) [inline outplacement_deviation_report/wizards/deviation_report_wizard.py:161]: Do we need to log the responses?

File: outplacement_deviationreport_ipf_client/models/client_config.py [hunk 4]
```diff
@@ -83,13 +84,14 @@ def create_request_history(self, method, url, response, payload=False,
...
+            values.update(message=json.loads(response.content))
         self.env['ipf.report.request.history'].create(values)
 
     def get_headers(self):
         tracking_id = pycompat.text_type(uuid.uuid1())
         headers = {
-            'x-amf-mediaType': "application/json",
+            'Content-Type': "application/json",
             'AF-TrackingId': tracking_id,
             'AF-SystemId': "AF-SystemId",
```
Comment (arturchilangwa-vertel) [inline outplacement_deviationreport_ipf_client/models/client_config.py]: The `AF-SystemId` looks suspicious, should be `AFDAFA` or `AFCRM`.

File: sale_outplacement/models/outplacement.py [hunk 5]
```diff
@@ -108,7 +105,8 @@ def _get_department_id(self, data):
         department = self.env['performing.operation'].search(
             [('ka_nr', '=', data.get('utforande_verksamhet_id', ''))],
             limit=1)
-        _logger.debug('Department: hr_department %s | %s' % (department, data.get('utforande_verksamhet_id')))
+        _logger.debug('Department: hr_department %s | %s' % (
+            department, data.get('utforande_verksamhet_id')))
```
Comment (arturchilangwa-vertel) [inline sale_outplacement/models/outplacement.py:109]: Do we need this logging?

**Review by arturchilangwa-vertel — CHANGES_REQUESTED** [review]

File: outplacement/models/outplacement.py (hunk 1 above)
Comment (nils-vertel) [reply outplacement/models/outplacement.py:228]: Same as for the whole block of comments, but I'll add a comment on this function as well

File: outplacement_deviation_report/wizards/deviation_report_wizard.py (hunk 3 above)
Comment (nils-vertel) [reply outplacement_deviation_report/wizards/deviation_report_wizard.py:161]: Yep, if something has gone wrong we want to see why, its only when something has gone wrong we log this

File: sale_outplacement/models/outplacement.py (hunk 5 above)
Comment (nils-vertel) [reply sale_outplacement/models/outplacement.py:109]: Maybe maybe not, its a debug logg so it does not cost anything unless debugging is set

**Review by arturchilangwa-vertel — APPROVED** [review]

**Review by arturchilangwa-vertel — APPROVED** [review]

**Comment by arturchilangwa-vertel** [comment]
Just going to try to install the modules on my machine.

**Comment by arturchilangwa-vertel** [comment]
Works to install on my machine!

**Pull request merged**